tests/test_ccs.py

```diff
@@ -0,0 +1,53 @@
+import unittest
+try:
+    from unittest.mock import patch, ANY
+except:
+    from mock import patch, ANY
+
+from tests.http_client import MinimalResponse
+from tests.test_token_cache import build_response
+
+import msal
+
+
+class TestCcsRoutingInfoTestCase(unittest.TestCase):
+
+    def test_acquire_token_by_auth_code_flow(self):
+        app = msal.ClientApplication("client_id")
+        state = "foo"
+        flow = app.initiate_auth_code_flow(
+            ["some", "scope"], login_hint="johndoe@contoso.com", state=state)
+        with patch.object(app.http_client, "post", return_value=MinimalResponse(
+                status_code=400, text='{"error": "mock"}')) as mocked_method:
+            app.acquire_token_by_auth_code_flow(flow, {"state": state, "code": "bar"})
+            self.assertEqual(
+                "UPN:johndoe@contoso.com",
+                mocked_method.call_args[1].get("headers", {}).get('X-AnchorMailbox'),
+                "CSS routing info should be derived from login_hint")
```

Comment:
Hi @trwalke , you have been one of the early adopters who implements the CCS Routing info workitem. Would you please help reviewing this PR? You do not have to review all those implementation code written in Python, but please give a high level confirmation on the behaviors demonstrated by the test cases.

This test case here indicates that, an auth code flow will add an `X-AnchorMailbox: UPN:johndoe@contoso.com` header.